crates/ironrdp-pdu/src/gcc/conference_create.rs

```diff
@@ -26,11 +26,34 @@ const CONFERENCE_NAME: &[u8] = b"1";
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct ConferenceCreateRequest {
-    pub gcc_blocks: ClientGccBlocks,
+    /// INVARIANT: `gcc_blocks.size() + CONFERENCE_REQUEST_CONNECT_PDU_SIZE <= u16::MAX`
+    gcc_blocks: ClientGccBlocks,
 }
 
 impl ConferenceCreateRequest {
     const NAME: &'static str = "ConferenceCreateRequest";
+
+    pub fn new(gcc_blocks: ClientGccBlocks) -> DecodeResult<Self> {
+        // Ensure the invariant on gcc_blocks.size() is respected.
+        check_invariant(gcc_blocks.size() + CONFERENCE_REQUEST_CONNECT_PDU_SIZE <= usize::from(u16::MAX)).ok_or_else(
+            || {
+                invalid_field_err!(
+                    "gcc_blocks",
+                    "gcc_blocks.size() + CONFERENCE_REQUEST_CONNECT_PDU_SIZE > u16::MAX"
+                )
+            },
+        )?;
```

Comment:
@RRRadicalEdward Here is how we can keep the invariant properly spelled in the check (https://github.com/Devolutions/IronRDP/pull/964#discussion_r2322718078)